client/src/explorer/model/RuntimeTreeItem.ts

```diff
@@ -21,24 +21,28 @@ import { FabricRuntimeManager } from '../../fabric/FabricRuntimeManager';
 import { FabricRuntime } from '../../fabric/FabricRuntime';
 
 export class RuntimeTreeItem extends ConnectionTreeItem {
+
+    static async newRuntimeTreeItem(provider: BlockchainExplorerProvider, label: string, connection: any, collapsableState: vscode.TreeItemCollapsibleState, command?: vscode.Command) {
+        const treeItem: RuntimeTreeItem = new RuntimeTreeItem(provider, label, connection, collapsableState);
+        await treeItem.updateProperties();
+        return treeItem;
+    }
+
     contextValue = 'blockchain-runtime-item';
 
     private name: string;
     private runtime: FabricRuntime;
     private busyTicker: NodeJS.Timer;
     private busyTicks: number = 0;
 
-    constructor(provider: BlockchainExplorerProvider, public readonly label: string, public readonly connection: any, public readonly collapsableState: vscode.TreeItemCollapsibleState, public readonly command?: vscode.Command) {
+    private constructor(provider: BlockchainExplorerProvider, public readonly label: string, public readonly connection: any, public readonly collapsableState: vscode.TreeItemCollapsibleState, public readonly command?: vscode.Command) {
```

Comment:
why do you need another function to call the constructor?

Reply:
Because we need to do async stuff when creating a new RuntimeTreeItem (determining the state of the runtime), and you can't do that in a constructor - that's why the newRuntimeTreeItem function is async and does additional stuff to new. The alternative is to move the additional stuff into a public `initialize` function and then hope everyone remembers to call it; at least this way they have no choice in the matter.